features/net/network-socket/WiFiInterface.h

```diff
@@ -17,22 +17,24 @@
 #ifndef WIFI_INTERFACE_H
 #define WIFI_INTERFACE_H
 
+#include <string.h>
 #include "network-socket/NetworkInterface.h"
 
+typedef struct wifi_ap {
+    char ssid[33]; /* 32 is what 802.11 defines as longest possible name; +1 for the \0 */
+    uint8_t bssid[6];
+    nsapi_security_t security;
+    int16_t rssi;
+    uint8_t channel;
 
-/** Enum of WiFi encryption types
- *
- *  The security type specifies a particular security to use when
- *  connected to a WiFi network
- *
- *  @enum nsapi_protocol_t
- */
-enum nsapi_security_t {
-    NSAPI_SECURITY_NONE = 0,   /*!< open access point */
-    NSAPI_SECURITY_WEP,        /*!< phrase conforms to WEP */
-    NSAPI_SECURITY_WPA,        /*!< phrase conforms to WPA */
-    NSAPI_SECURITY_WPA2,       /*!< phrase conforms to WPA2 */
-};
+    wifi_ap()
+    {
+        memset(this, 0, sizeof(*this));
+    }
+} wifi_ap_t;
+
+typedef void (*wifi_ap_scan_cb_t)(wifi_ap_t *ap, void *data);
+typedef void (*wifi_connect_cb_t)(nsapi_error_t res, void *data);
```

Comment:
Small though: flipping these would allow users to take advantage of the callback's `thunk` function:

```
Callback<void(wifi_ap_t *ap)> user_callback(&object, &Object::doit);
typedef void (*wifi_ap_scan_cb_t)(wifi_ap_t *ap, void *data);

wifi_ap_scan_cb_t cb = &Callback<void(wifi_ap_t *)>::thunk
void *data = &user_callback;

cb(data);
```

Actually, would it be possible to just adopt the Callback class in this API?